pkg/theia/commands/clickhouse_status.go

```diff
@@ -0,0 +1,271 @@
+}
+
+var options *chOptions
+
+var clickHouseStatusCmd = &cobra.Command{
+	Use:     "status",
+	Short:   "Get diagnostic infos of ClickHouse database",
+	Example: example,
+	Args:    cobra.NoArgs,
+	RunE:    getClickHouseStatus,
+}
+
+var example = strings.Trim(`
+theia clickhouse status --diskInfo
+theia clickhouse status --diskInfo --tableInfo
+theia clickhouse status --diskInfo --tableInfo --insertRate
+`, "\n")
+
+func init() {
+	clickHouseCmd.AddCommand(clickHouseStatusCmd)
+	options = &chOptions{}
+	clickHouseStatusCmd.Flags().BoolVar(&options.diskInfo, "diskInfo", false, "check disk usage information")
+	clickHouseStatusCmd.Flags().BoolVar(&options.tableInfo, "tableInfo", false, "check basic table information")
+	clickHouseStatusCmd.Flags().BoolVar(&options.insertRate, "insertRate", false, "check the insertion-rate of clickhouse")
+	clickHouseStatusCmd.Flags().BoolVar(&options.stackTraces, "stackTraces", false, "check stacktrace of clickhouse")
+}
+
+func getClickHouseStatus(cmd *cobra.Command, args []string) error {
+	if !options.diskInfo && !options.tableInfo && !options.insertRate && !options.stackTraces {
+		return fmt.Errorf("no metric related flag is specified")
+	}
+	kubeconfig, err := ResolveKubeConfig(cmd)
+	if err != nil {
+		return err
+	}
+	clientset, err := CreateK8sClient(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("couldn't create k8s client using given kubeconfig, %v", err)
+	}
+
+	endpoint, err := cmd.Flags().GetString("clickhouse-endpoint")
+	if err != nil {
+		return err
+	}
+	if endpoint != "" {
+		_, err := url.ParseRequestURI(endpoint)
+		if err != nil {
+			return fmt.Errorf("failed to decode input endpoint %s into a url, err: %v", endpoint, err)
+		}
+	}
+	useClusterIP, err := cmd.Flags().GetBool("use-cluster-ip")
+	if err != nil {
+		return err
+	}
+	if err := CheckClickHousePod(clientset); err != nil {
+		return err
+	}
+	// Connect to ClickHouse and get the result
+	connect, pf, err := SetupClickHouseConnection(clientset, kubeconfig, endpoint, useClusterIP)
+	if err != nil {
+		return err
+	}
+	if pf != nil {
+		defer pf.Stop()
+	}
+	if options.diskInfo {
+		data, err := getDataFromClickHouse(connect, diskQuery)
+		if err != nil {
+			return err
+		}
+		TableOutput(data)
+	}
+	if options.tableInfo {
+		data, err := getDataFromClickHouse(connect, tableInfoQuery)
+		if err != nil {
+			return err
+		}
+		TableOutput(data)
+	}
+	if options.insertRate {
+		data, err := getDataFromClickHouse(connect, insertRateQuery)
+		if err != nil {
+			return err
+		}
+		TableOutput(data)
+	}
+	if options.stackTraces {
+		data, err := getDataFromClickHouse(connect, stackTracesQuery)
+		if err != nil {
+			return err
+		}
+		TableOutputVertical(data)
+	}
+	return nil
+}
+
+func getDataFromClickHouse(connect *sql.DB, query int) ([][]string, error) {
+	result, err := connect.Query(queryMap[query])
+	if err != nil {
+		return nil, fmt.Errorf("failed to get clickhouse stack trace: %v", err)
```

Comment:
This error message also needs an update.

Reply:
Fixed. Thanks.